cron/monthly/send-monthly-report.js

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env node
+
+process.env.PORT = 3066;
+
+import _ from 'lodash';
+import app  from '../../index';
+import moment from 'moment';
+import config from 'config';
+
+const emailLib = require('../../server/lib/email')(app);
+const debug = require('debug')('monthlyreport');
+
+const Group = app.set('models').Group;
+const User = app.set('models').User;
+const Notification = app.set('models').Notification;
+
+const processGroups = (groups) => {
+    const promises = [];
+    groups.map(g => {
+        promises.push(processGroup(g));
+    })
+    return Promise.all(promises);
+};
+
+const d = new Date;
+d.setMonth(d.getMonth() - 1);
+const month = moment(d).format('MMMM');
+
+const init = () => {
+
+  const query = {
+      attributes: [
+          'id',
+          'slug',
+          'name',
+          'currency'
+      ],
+      where: {
+        slug: { $in: ['railsgirlsatl'] } // for testing
```

Comment:
What's the need for leaving that slug in there?

On Fri, Aug 5, 2016 at 2:51 PM Xavier Damman <notifications@github.com>
wrote:

> In cron/monthly/send-monthly-report.js
> <https://github.com/OpenCollective/opencollective-api/pull/516#discussion_r73742332>
> :
>
> > +
> > +const d = new Date;
> > +d.setMonth(d.getMonth() - 1);
> > +const month = moment(d).format('MMMM');
> > +
> > +const init = () => {
> > +
> > +  const query = {
> > +      attributes: [
> > +          'id',
> > +          'slug',
> > +          'name',
> > +          'currency'
> > +      ],
> > +      where: {
> > +        slug: { $in: ['railsgirlsatl'] } // for testing
>
> It is done through the notification table.
> But it is first fetching all the groups, then for each group looking for
> who to notify in the Notifications table.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenCollective/opencollective-api/pull/516/files/a062fd4540338c183dc0ca75be1e53c0f11f3ad8#r73742332>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF0vPUl4-NYg4W4WPaevKK_wghOFbjeSks5qc4YvgaJpZM4Jd4YS>
> .
>